src/core/src/runtime/tensor.cpp

```diff
@@ -178,6 +178,14 @@ void read_tensor_data(const std::filesystem::path& file_name, Tensor& tensor, si
     std::ifstream fin(file_name, std::ios::binary);
     fin.seekg(offset);
     auto bytes_to_read = tensor.get_byte_size();
+
+    OPENVINO_ASSERT(bytes_to_read <= static_cast<size_t>(std::numeric_limits<std::streamsize>::max()),
```

Comment:
Just to consider (as separate PR) if is valid scenario.
The `bytes_to_read` could be min of tensor byte size and streamsize limit.

If tensor is bigger then streamsize limit, then stream can be read into tensor.